lib/src/main/java/io/jenkins/tools/incrementals/lib/UpdateChecker.java

```diff
@@ -94,6 +97,18 @@ public String fullURL(String type) {
         }
     }
 
+    public @CheckForNull String findGroupId(String artifactId) throws IOException, InterruptedException {
+        String cacheKey = artifactId;
+        if (groupIdCache.containsKey(cacheKey)) {
+            log.info("Group ID Cache hit on artifact ID: " + artifactId);
+            return groupIdCache.get(cacheKey);
+        }
+
+        //TODO: implement to support non-Incremental formats
+        // Needs to load UC JSON and query it like https://github.com/jenkinsci/docker/pull/668
```

Comment:
TODO for me. Will implement if the PR is considered as feasible

Reply:
definitely feasible